pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,274 @@
+// validate 'UserDeleted' on channel 'user-events' with operation 'publish'
+va.validate('UserDeleted', {
+  userId: '123456789',
+  userEmail: 'alex@mail.com',
+}, 'user-events', 'publish')`}
+</CodeBlock>
+
+# Validation using AsyncAPI gateway
+
+AsyncAPI gateway intercepts all incoming messages moving them into a pipeline of middlewares and handlers such as Message validator.As per today, only the Kafka protocol is supported.
+
+# Here's an example usecase using AsyncAPI gateway
+
+1. Let's create the AsyncAPI file. Expected messages are based on a small portion of the StreetLights tutorial.
+
+<CodeBlock language="yaml">
+{`asyncapi: '2.4.0'
+info:
+  title: AsyncAPI Event-Gateway demo API
+  version: 1.0.0-alpha
+  description: This API lets users interact with an instance of the [AsyncAPI Event-Gateway](https://github.com/asyncapi/event-gateway).
+defaultContentType: application/json
+servers:
+  asyncapi-event-gateway-demo:
+    url: 'event-gateway-demo.asyncapi.com:20472'
+    protocol: kafka
+    description: AsyncAPI [Event-Gateway](https://github.com/asyncapi/event-gateway) demo Kafka proxy. Expected messages are based on a small portion of the [StreetLights tutorial](https://bit.ly/asyncapi).
+  asyncapi-event-gateway-demo-validation:
+    url: 'event-gateway-demo.asyncapi.com:5000/ws'
+    protocol: ws
+    description: AsyncAPI [Event-Gateway](https://github.com/asyncapi/event-gateway) demo. Subscribe for Kafka proxy message validation errors.
+  asyncapi-kafka-test:
+    url: 'asyncapi-kafka-test-asyncapi-8f90.aivencloud.com:20472' # Kafka with 3 brokers.
+    protocol: kafka-secure
+    description: AsyncAPI Kafka test broker. Private.
+channels:
+  event-gateway-demo:
+    description: Demo Kafka topic for asyncapi-event-gateway-demo server. Users can send their events to this topic and see how message validation happens on the fly based on this right AsyncAPI file by connecting to `event-gateway-demo-validation-events` channel (`asyncapi-event-gateway-demo-validation` ws server).
+    x-servers: # Based on https://github.com/asyncapi/spec/pull/531
+      - asyncapi-event-gateway-demo
+    publish:
+      message:
+        $ref: '#/components/messages/lightMeasured'
+  event-gateway-demo-validation-events:
+    description: Validation errors are published here, so users can see how message validation happens on the fly based on this right AsyncAPI file.
+    x-servers: # Based on https://github.com/asyncapi/spec/pull/531
+      - asyncapi-event-gateway-demo-validation
+    subscribe:
+      message:
+        $ref: '#/components/messages/invalidMessage'
+  event-gateway-demo-validation:
+    description: Validation errors are published to and consumed from it. AsyncAPI Event-gateway is the only user of this channel. It can be consumed and exposed via `event-gateway-demo-validation-events` channel (`asyncapi-event-gateway-demo-validation` ws server).
```

Comment:
```suggestion
    description: Validation errors are published to and consumed from it. AsyncAPI Event-gateway is the only user of this channel. It can be consumed and exposed via \`event-gateway-demo-validation-events\` channel (\`asyncapi-event-gateway-demo-validation\` ws server).
```